test/Rings/solving-test.jl

```diff
@@ -6,8 +6,16 @@
     denom = 336*x^3 - 120*x^2 + 2*x + 1
     p1  = -184*x^3 + 80*x^2 - 4*x - 1
     p2 = -36*x^3 + 18*x^2 - 2*x
-    sols = Vector{fmpq}[[744483363399261433351//1180591620717411303424, 372241681699630716673//1180591620717411303424, -154187553040555781639//1180591620717411303424], [1, 0, 0], [71793683196126133110381699745//316912650057057350374175801344, 71793683196126133110381699745//633825300114114700748351602688, 173325283664805084153412401855//633825300114114700748351602688], [196765270119568550571//590295810358705651712, 1//590295810358705651712, 196765270119568550571//590295810358705651712]]
-    res = real_solutions(I)
+    prec = [32,48,64]
+    for p in prec
+        res = real_solutions(I, precision=p)
+        sols = res[1]
+        for g in gens(I)
+            for s in sols
+                @test abs(evaluate(g, s)) < fmpq(2)^(-p+1)
+            end
+        end
+    end
     @test res[1] == sols
```

Comment:
outdated line? you are setting sols to res[1] just above it inside the new loop.